### PR TITLE
[3006.x] Only generate the HMAC's for ``libssl.so.1.1`` and ``libcrypto.so.1.1`` if those files exist

### LINKED ISSUE #65581: [BUG] Missing File when install Minion 3006.X
**Description**
When installing via Bootstrap.sh on a Rocky8 instance using a onedir install I am getting errors that files are missing. This is happening in AWS
```
/opt/saltstack/salt/lib/libssl.so.1.1: No such file or directory
/opt/saltstack/salt/lib/libcrypto.so.1.1: No such file or directory
```

**Setup**
```
wget -O bootstrap-salt.sh https://bootstrap.saltstack.com
sudo sh bootstrap-salt.sh -X -P onedir
```

Please be as specific as possible and give set-up details.

- [ ] on-prem machine
- [ ] VM (Virtualbox, KVM, etc. please specify)
- [X] VM running on a cloud service, please be explicit and add details
- [ ] container (Kubernetes, Docker, containerd, etc. please specify)
- [ ] or a combination, please be explicit
- [ ] jails if it is FreeBSD
- [ ] classic packaging
- [X] onedir packaging
- [X] used bootstrap to install

**Steps to Reproduce the behavior**
ROCKY8 AWS AMI ami-097104a26f5e1c26a

**Expected behavior**
Install to run cleanly with no errors. 

**Screenshots**
```
 *  INFO: Running install_rocky_linux_onedir()
Last metadata expiration check: 0:00:02 ago on Wed 23 Aug 2023 10:14:20 AM EDT.
Dependencies resolved.
================================================================================
 Package             Architecture   Version             Repository         Size
================================================================================
Installing:
 salt-minion         x86_64         3006.2-0            saltstack         215 k
Installing dependencies:
 pciutils            x86_64         3.7.0-3.el8         baseos            104 k
 salt                x86_64         3006.2-0            saltstack          58 M

Transaction Summary
================================================================================
Install  3 Packages

Total download size: 59 M
Installed size: 175 M
Downloading Packages:
(1/3): pciutils-3.7.0-3.el8.x86_64.rpm          352 kB/s | 104 kB     00:00    
(2/3): salt-minion-3006.2-0.x86_64.rpm          617 kB/s | 215 kB     00:00    
(3/3): salt-3006.2-0.x86_64.rpm                  70 MB/s |  58 MB     00:00    
--------------------------------------------------------------------------------
Total                                            64 MB/s |  59 MB     00:00     
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                        1/1 
  Installing       : pciutils-3.7.0-3.el8.x86_64                            1/3 
  Running scriptlet: salt-3006.2-0.x86_64                                   2/3 
  Installing       : salt-3006.2-0.x86_64                                   2/3 
  Running scriptlet: salt-3006.2-0.x86_64                                   2/3 
  Installing       : salt-minion-3006.2-0.x86_64                            3/3 
  Running scriptlet: salt-minion-3006.2-0.x86_64                            3/3 
/opt/saltstack/salt/lib/libssl.so.1.1: No such file or directory
/opt/saltstack/salt/lib/libcrypto.so.1.1: No such file or directory

  Verifying        : pciutils-3.7.0-3.el8.x86_64                            1/3 
  Verifying        : salt-3006.2-0.x86_64                                   2/3 
  Verifying        : salt-minion-3006.2-0.x86_64                            3/3 

Installed:
  pciutils-3.7.0-3.el8.x86_64 salt-3006.2-0.x86_64 salt-minion-3006.2-0.x86_64

Complete!
 *  INFO: Running install_rocky_linux_onedir_post()
 *  INFO: Running install_rocky_linux_check_services()
 *  INFO: Salt installed!
```

```
[root@ip-10-2-7-37 ~]# ll /opt/saltstack/salt/lib/libssl.so.1.1
ls: cannot access '/opt/saltstack/salt/lib/libssl.so.1.1': No such file or directory
[root@ip-10-2-7-37 ~]# ll /opt/saltstack/salt/lib/libcrypto.so.1.1
ls: cannot access '/opt/saltstack/salt/lib/libcrypto.so.1.1': No such file or directory
```

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.)

```yaml
[root@ip-10-2-7-37 ~]# salt-minion --versions-report
Salt Version:
          Salt: 3006.2
 
Python Version:
        Python: 3.10.12 (main, Aug  3 2023, 21:47:10) [GCC 11.2.0]
 
Dependency Versions:
          cffi: 1.14.6
      cherrypy: 18.6.1
      dateutil: 2.8.1
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 3.1.2
       libgit2: Not Installed
  looseversion: 1.0.2
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     packaging: 22.0
     pycparser: 2.21
      pycrypto: Not Installed
  pycryptodome: 3.9.8
        pygit2: Not Installed
  python-gnupg: 0.4.8
        PyYAML: 6.0.1
         PyZMQ: 23.2.0
        relenv: 0.13.3
         smmap: Not Installed
       timelib: 0.2.4
       Tornado: 4.5.3
           ZMQ: 4.3.4
 
System Versions:
          dist: rocky 8.8 Green Obsidian
        locale: utf-8
       machine: x86_64
       release: 4.18.0-477.21.1.el8_8.x86_64
        system: Linux
       version: Rocky Linux 8.8 Green Obsidian

```
</details>

**Additional context**
When searching the issues in git noticed the following issue:
[BUG] Install / Upgrade to 3006 RC 1 on Centosstream 8 showing issues saltstack/salt#63821

which seems to have a sorta similar problem

### PR DESCRIPTION
**Pull request by s0undt3ch**
### What does this PR do?
See title

### What issues does this PR fix or reference?
Fixes #65581

### CODE REVIEW

**Review by twangboy — APPROVED** [review]

**Review by dwoz — APPROVED** [review]

File: pkg/rpm/salt.spec [hunk 1]
```diff
@@ -482,8 +486,12 @@ if [ $1 -lt 2 ]; then
   # ensure hmac are up to date, master or minion, rest install one or the other
   # key used is from openssl/crypto/fips/fips_standalone_hmac.c openssl 1.1.1k
   if [ $(cat /etc/os-release | grep VERSION_ID | cut -d '=' -f 2 | sed  's/\"//g' | cut -d '.' -f 1) = "8" ]; then
```
Comment (dmurphy18) [inline pkg/rpm/salt.spec:488]: you need this line, since only need it on RedHat 8, e.g. RHEL 9 doesn't use 1.1.1

**Review by dmurphy18 — COMMENTED** [review]
You need the 8 since only needed on RHEL 8

File: pkg/rpm/salt.spec (hunk 1 above)
Comment (s0undt3ch) [reply pkg/rpm/salt.spec:488]: I'm not taking it out

Comment (dmurphy18) [reply pkg/rpm/salt.spec:488]: d'oh

**Pull request merged**